contracts/HasIntents.sol

```diff
@@ -0,0 +1,32 @@
+/* solhint-disable-next-line compiler-fixed */
+pragma solidity ^0.4.23;
+
+// Copyright 2018 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+// Common: HasIntents
+//
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
+
+
+contract HasIntents {
+    /*
+     * Mappings
+     */
+    mapping(bytes32 /* intentHash */ => bytes32) internal intents;
```

Comment:
Just my views: 

I believe that this will be accessed from outside, or interaction layer can call this from outside to check if the value is stored against the key, before generating the proof data.
Hmm...not sure as of now if this should be public or internal. May be this can be changed if needed as we work upon later.

Reply:
I kept it internal for the reason that I only expected it to be used by the OpenSTValue contract, but if this will be called on directly from the interaction layer as you have pointed out, then it makes sense to change its type to Public. Will do.